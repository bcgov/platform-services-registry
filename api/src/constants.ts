//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

export const ENVIRONMENTS = {
  DEVELOPMENT: 'development',
  PRODUCTION: 'production',
};

export const projectSetNames = [
  'tools',
  'dev',
  'test',
  'prod',
];

export const ROLE_IDS = {
  PRODUCT_OWNER: 1,
  TECHNICAL_CONTACT: 2,
};

export const USER_ROLES = {
  ADMINISTRATOR: 'administrator',
};

export const WEB_CLIENT_ID = 'registry-web';
export const API_CLIENT_ID = 'registry-api';
export const BOT_CLIENT_ID = 'registry-gitops-ci';

export const STATUS_ERROR = {};
STATUS_ERROR[401] = 'Unauthorized';
STATUS_ERROR[500] = 'Internal Server Error';

export const CLUSTER_NAMES = [
  'silver',
  'gold',
  'golddr',
  'klab',
  'clab',
]

export const GOLD_QUORUM_COUNT = 2;