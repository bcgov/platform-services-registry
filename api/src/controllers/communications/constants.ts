/* eslint-disable import/prefer-default-export */
export const SELECTEDCOMMUNICATIONS = [
  {
    segmentID: "5",
    segmentName: "Pathfinder SSO Segment",
    description:
      "This mailing list includes updates about major upcoming changes for the BC Gov's Pathfinder Single Sign On (SSO) Service (KeyCloak), important information for the product teams, and updates about SSO Service outages",
    isChecked: true,
  },
  {
    segmentID: "7",
    segmentName: "Platform Services team - test",
    description: "This is for testing your email before sending it!",
    isChecked: true,
  },
  {
    segmentID: "1",
    segmentName: "Platform Services Updates",
    description:
      "This mailing list includes updates about major upcoming changes on the Platform, important information for the product teams, and updates about Platform outages",
    isChecked: true,
  },
  {
    segmentID: "4",
    segmentName: "SSO Updates",
    description:
      "This mailing list includes updates about major upcoming changes for the BC Gov's Pathfinder Single Sign On (SSO) Service (KeyCloak), important information for the product teams, and updates about SSO Service outages",
    isChecked: true,
  },
  {
    segmentID: "6",
    segmentName: "SSO Updates Core Team",
    description: "This mailing list goes to the Pathfinder SSO core team&nbsp;",
    isChecked: true,
  },
];
