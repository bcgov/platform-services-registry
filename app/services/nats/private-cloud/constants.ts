import { CpuQuota, MemoryQuota, StorageQuota } from '@/validation-schemas/private-cloud';

export const Cluster = {
  CLAB: 'clab',
  KLAB: 'klab',
  SILVER: 'silver',
  GOLD: 'gold',
  GOLDDR: 'golddr',
  KLAB2: 'klab2',
  EMERALD: 'emerald',
};

export const RequestType = {
  CREATE: 'create',
  EDIT: 'edit',
  DELETE: 'delete',
};

export interface CpuOption {
  name: string;
  cpuRequests: number;
  cpuLimits: number;
}

export type DefaultCpuOptionsKey = (typeof CpuQuota)['_type'];
export type DefaultMemoryOptionsKey = (typeof MemoryQuota)['_type'];
export type DefaultStorageOptionsKey = (typeof StorageQuota)['_type'];

export const DefaultCpuOptions: Record<DefaultCpuOptionsKey, CpuOption> = {
  CPU_REQUEST_0_5_LIMIT_1_5: {
    name: 'cpu-request-0.5-limit-1.5',
    cpuRequests: 0.5,
    cpuLimits: 1.5,
  },
  CPU_REQUEST_1_LIMIT_2: {
    name: 'cpu-request-1-limit-2',
    cpuRequests: 1,
    cpuLimits: 2,
  },
  CPU_REQUEST_2_LIMIT_4: {
    name: 'cpu-request-2-limit-4',
    cpuRequests: 2,
    cpuLimits: 4,
  },
  CPU_REQUEST_4_LIMIT_8: {
    name: 'cpu-request-4-limit-8',
    cpuRequests: 4,
    cpuLimits: 8,
  },
  CPU_REQUEST_8_LIMIT_16: {
    name: 'cpu-request-8-limit-16',
    cpuRequests: 8,
    cpuLimits: 16,
  },
  CPU_REQUEST_16_LIMIT_32: {
    name: 'cpu-request-16-limit-32',
    cpuRequests: 16,
    cpuLimits: 32,
  },
  CPU_REQUEST_32_LIMIT_64: {
    name: 'cpu-request-32-limit-64',
    cpuRequests: 32,
    cpuLimits: 64,
  },
  CPU_REQUEST_64_LIMIT_128: {
    name: 'cpu-request-64-limit-128',
    cpuRequests: 64,
    cpuLimits: 128,
  },
};

export interface MemoryOption {
  name: string;
  memoryRequests: string;
  memoryLimits: string;
}

export const DefaultMemoryOptions: Record<DefaultMemoryOptionsKey, MemoryOption> = {
  MEMORY_REQUEST_2_LIMIT_4: {
    name: 'memory-request-2-limit-4',
    memoryRequests: '2Gi',
    memoryLimits: '4Gi',
  },
  MEMORY_REQUEST_4_LIMIT_8: {
    name: 'memory-request-4-limit-8',
    memoryRequests: '4Gi',
    memoryLimits: '8Gi',
  },
  MEMORY_REQUEST_8_LIMIT_16: {
    name: 'memory-request-8-limit-16',
    memoryRequests: '8Gi',
    memoryLimits: '16Gi',
  },
  MEMORY_REQUEST_16_LIMIT_32: {
    name: 'memory-request-16-limit-32',
    memoryRequests: '16Gi',
    memoryLimits: '32Gi',
  },
  MEMORY_REQUEST_32_LIMIT_64: {
    name: 'memory-request-32-limit-64',
    memoryRequests: '32Gi',
    memoryLimits: '64Gi',
  },
  MEMORY_REQUEST_64_LIMIT_128: {
    name: 'memory-request-64-limit-128',
    memoryRequests: '64Gi',
    memoryLimits: '128Gi',
  },
  MEMORY_REQUEST_128_LIMIT_256: {
    name: 'memory-request-128-limit-256',
    memoryRequests: '128Gi',
    memoryLimits: '256Gi',
  },
};

export interface StorageOption {
  name: string;
  storagePvcCount: 60;
  storageFile: string;
  storageBackup: string;
  storageCapacity: string;
  storageBlock: string;
}

export const DefaultStorageOptions: Record<DefaultStorageOptionsKey, StorageOption> = {
  STORAGE_1: {
    name: 'storage-1',
    storagePvcCount: 60,
    storageFile: '1Gi',
    storageBackup: '1Gi',
    storageCapacity: '1Gi',
    storageBlock: '1Gi',
  },
  STORAGE_2: {
    name: 'storage-2',
    storagePvcCount: 60,
    storageFile: '2Gi',
    storageBackup: '1Gi',
    storageCapacity: '2Gi',
    storageBlock: '2Gi',
  },
  STORAGE_4: {
    name: 'storage-4',
    storagePvcCount: 60,
    storageFile: '4Gi',
    storageBackup: '2Gi',
    storageCapacity: '4Gi',
    storageBlock: '4Gi',
  },
  STORAGE_16: {
    name: 'storage-16',
    storagePvcCount: 60,
    storageFile: '16Gi',
    storageBackup: '8Gi',
    storageCapacity: '16Gi',
    storageBlock: '16Gi',
  },
  STORAGE_32: {
    name: 'storage-32',
    storagePvcCount: 60,
    storageFile: '32Gi',
    storageBackup: '16Gi',
    storageCapacity: '32Gi',
    storageBlock: '32Gi',
  },
  STORAGE_64: {
    name: 'storage-64',
    storagePvcCount: 60,
    storageFile: '64Gi',
    storageBackup: '32Gi',
    storageCapacity: '64Gi',
    storageBlock: '64Gi',
  },
  STORAGE_128: {
    name: 'storage-128',
    storagePvcCount: 60,
    storageFile: '128Gi',
    storageBackup: '64Gi',
    storageCapacity: '128Gi',
    storageBlock: '128Gi',
  },
  STORAGE_256: {
    name: 'storage-256',
    storagePvcCount: 60,
    storageFile: '256Gi',
    storageBackup: '128Gi',
    storageCapacity: '256Gi',
    storageBlock: '256Gi',
  },
  STORAGE_512: {
    name: 'storage-512',
    storagePvcCount: 60,
    storageFile: '512Gi',
    storageBackup: '256Gi',
    storageCapacity: '512Gi',
    storageBlock: '512Gi',
  },
};

export const snapshot = {
  name: 'snapshot-5',
  snapshotCount: 5,
};

export type QuotaOption = {
  cpu: CpuOption;
  memory: MemoryOption;
  storage: StorageOption;
};

export const testMessage = {
  action: 'create',
  profile_id: 6,
  cluster_id: 1,
  cluster_name: 'clab',
  display_name: 'Oamar test',
  description: 'asdf',
  ministry_id: 'AGRI',
  merge_type: 'auto',
  namespaces: [
    {
      namespace_id: 21,
      name: 'dad3d6-tools',
      quota: {
        cpu: 'cpu-request-0.5-limit-1.5',
        memory: 'memory-request-2-limit-4',
        storage: 'storage-1', // the one refers to the file
        snapshot: 'snapshot-5',
      },
      quotas: {
        cpu: { requests: 0.5, limits: 1.5 },
        memory: { requests: '2Gi', limits: '4Gi' },
        storage: {
          block: '1Gi',
          file: '1Gi',
          backup: '512Mi',
          capacity: '1Gi',
          pvc_count: 60,
        },
        snapshot: { count: 5 },
      },
    },
    {
      namespace_id: 22,
      name: 'dad3d6-dev',
      quota: {
        cpu: 'cpu-request-0.5-limit-1.5',
        memory: 'memory-request-2-limit-4',
        storage: 'storage-1',
        snapshot: 'snapshot-5',
      },
      quotas: {
        cpu: { requests: 0.5, limits: 1.5 },
        memory: { requests: '2Gi', limits: '4Gi' },
        storage: {
          block: '1Gi',
          file: '1Gi',
          backup: '512Mi',
          capacity: '1Gi',
          pvc_count: 60,
        },
        snapshot: { count: 5 },
      },
    },
    {
      namespace_id: 23,
      name: 'dad3d6-test',
      quota: {
        cpu: 'cpu-request-0.5-limit-1.5',
        memory: 'memory-request-2-limit-4',
        storage: 'storage-1',
        snapshot: 'snapshot-5',
      },
      quotas: {
        cpu: { requests: 0.5, limits: 1.5 },
        memory: { requests: '2Gi', limits: '4Gi' },
        storage: {
          block: '1Gi',
          file: '1Gi',
          backup: '512Mi',
          capacity: '1Gi',
          pvc_count: 60,
        },
        snapshot: { count: 5 },
      },
    },
    {
      namespace_id: 24,
      name: 'dad3d6-prod',
      quota: {
        cpu: 'cpu-request-0.5-limit-1.5',
        memory: 'memory-request-2-limit-4',
        storage: 'storage-1',
        snapshot: 'snapshot-5',
      },
      quotas: {
        cpu: { requests: 0.5, limits: 1.5 },
        memory: { requests: '2Gi', limits: '4Gi' },
        storage: {
          block: '1Gi',
          file: '1Gi',
          backup: '512Mi',
          capacity: '1Gi',
          pvc_count: 60,
        },
        snapshot: { count: 5 },
      },
    },
  ],
  contacts: [
    {
      user_id: 'okanji',
      provider: 'github',
      email: 'oamar.kanji@gov.bc.ca',
      rocketchat_username: null,
      role: 'lead',
    },
    {
      user_id: 'w8896699',
      provider: 'github',
      email: 'billy.li@gov.bc.ca',
      rocketchat_username: null,
      role: 'owner',
    },
  ],
};
