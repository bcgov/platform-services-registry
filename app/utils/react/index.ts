export * from './hooks';
export * from './search-params';
