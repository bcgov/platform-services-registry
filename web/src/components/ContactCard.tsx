//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import React from 'react';
import { Box, Flex, Text } from 'rebass';
import theme from '../theme';

interface IContactCardProps {
  POName?: string;
  POEmail?: string;
  TCName?: string;
  TCEmail?: string;
};

const ContactCard: React.FC<IContactCardProps> = (props) => {
  const { POName = '', POEmail = '', TCName = '', TCEmail = '' } = props;

  return (
    <Flex flexWrap='wrap'>
        <Box width={1/2} px={2}>
            <Text as="h3" mt={4}>
                Product Owner:
            </Text>
        </Box>
        <Box width={1/2} px={2}> 
            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={2}>
                {POName}
            </Text>
            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={2}>
                {POEmail}
            </Text>
        </Box>
        <Box width={1/2} px={2}>
            <Text as="h3" mt={4}>
                Technical Contact:
            </Text>
        </Box>
        <Box width={1/2} px={2}> 
            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={2}>
                {TCName}
            </Text>
            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={2}>
                {TCEmail}
            </Text>
        </Box>
    </Flex>
  );
};

export default ContactCard;