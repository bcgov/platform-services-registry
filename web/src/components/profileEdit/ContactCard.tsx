//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import React from 'react';
import { Box, Flex, Text } from 'rebass';
import theme from '../../theme';

interface IContactCardProps {
    POName?: string;
    POEmail?: string;
    POGithubId?: string;
    TCName?: string;
    TCEmail?: string;
    TCGithubId?: string;
};

const ContactCard: React.FC<IContactCardProps> = (props) => {
    const { POName = '', POEmail = '', POGithubId = '', TCName = '', TCEmail = '', TCGithubId = '' } = props;

    return (
        <Flex flexWrap='wrap'>
            <Box width={1 / 2} px={2} mt={3}>
                <Text as="h3">
                    Product Owner
            </Text>
                <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
                    This is typically the business owner of the application;
            </Text>
            </Box>
            <Box width={1 / 2} px={2} mt={3}>
                <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
                    {POName}
                </Text>
                <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
                    {POEmail}
                </Text>
                <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
                    {POGithubId}
                </Text>
            </Box>
            <Box width={1 / 2} px={2} mt={3}>
                <Text as="h3">
                    Technical Contact
            </Text>
                <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
                    This is typically the DevOps specialist;
            </Text>
            </Box>
            <Box width={1 / 2} px={2} mt={3}>
                <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
                    {TCName}
                </Text>
                <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
                    {TCEmail}
                </Text>
                <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
                    {TCGithubId}
                </Text>
            </Box>
        </Flex>
    );
};

export default ContactCard;
