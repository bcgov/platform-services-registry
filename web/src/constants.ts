//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Jason Leach on 2020-06-05.
//

export const SSO_BASE_URL = process.env.NODE_ENV === 'production'
  ? '{{env "SSO_BASE_URL"}}'
  : 'https://sso-dev.pathfinder.gov.bc.ca';
export const SSO_CLIENT_ID = 'registry-web';
export const SSO_REALM_NAME = 'devhub';

export const API = {
  BASE_URL: () =>
    process.env.NODE_ENV === 'development'
      ? 'http://localhost:8100/api/v1/'
      : `${window.location.origin}/api/v1/`,
};

export const ROLES = {
  PRODUCTOWNER: 1,
  TECHNICAL: 2,
};

export const DEFAULT_MINISTRY = 'CITZ';

export const LAYOUT_SET_MIN = 'min';
export const LAYOUT_SET_AUTH = 'auth';
export const LAYOUT_SET_UNAUTH = 'unauth';

export const COMPONENT_METADATA = [
  { displayName: 'Notification: Email', inputValue: 'notificationEmail' },
  { displayName: 'Notification: SMS', inputValue: 'notificationSms' },
  { displayName: 'Notification: MS Teams', inputValue: 'notificationMsTeams' },
  { displayName: 'Payment processing: Bambora', inputValue: 'paymentBambora' },
  { displayName: 'Payment processing: PayBC', inputValue: 'paymentPayBc' },
  { displayName: 'File Transfer', inputValue: 'fileTransfer' },
  { displayName: 'File Storage', inputValue: 'fileStorage' },
  { displayName: 'Geo Mapping: Web-based', inputValue: 'geoMappingWeb' },
  { displayName: 'Geo Mapping: Location Services', inputValue: 'geoMappingLocation' },
  { displayName: 'Scheduling: Calendar', inputValue: 'schedulingCalendar' },
  { displayName: 'Scheduling: Appointments', inputValue: 'schedulingAppointments' },
  { displayName: 'Identity Management: SiteMinder', inputValue: 'idmSiteMinder' },
  { displayName: 'Identity Management: KeyCloak', inputValue: 'idmKeycloak' },
  { displayName: 'Identity Management: Active Directory', inputValue: 'idmActiveDir' }
]

export const ROUTE_PATHS = {
  DASHBOARD: '/dashboard',
  NOT_FOUND: '/page-not-found',
  LANDING: '/public-landing',
  FORM: '/projects/create'
};

export const HOME_PAGE_URL = ROUTE_PATHS.DASHBOARD;