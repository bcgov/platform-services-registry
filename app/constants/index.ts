import { Cluster, Ministry, Prisma, Provider } from '@prisma/client';
import { CpuQuotaEnum, MemoryQuotaEnum, StorageQuotaEnum } from '@/validation-schemas/private-cloud';

export const clusters = Object.values(Cluster).filter((cluster) => cluster !== 'GOLDDR');

export const ministries = Object.values(Ministry);

export const providers = Object.values(Provider);

export const providerOptions = providers.map((v) => ({
  label: v === Provider.AZURE ? 'MS Azure' : v,
  value: v,
}));

export const ministryOptions = [
  {
    value: 'AEST',
    label: 'Post-Secondary Education and Future Skills Contacts',
  },
  {
    value: 'AG',
    label: 'Attorney General',
  },
  {
    value: 'AGRI',
    label: 'Agriculture and Food',
  },
  {
    value: 'ALC',
    label: 'Advisory Committee Revitalization',
  },
  {
    value: 'BCPC',
    label: 'British Columbia Provincial Committee',
  },
  {
    value: 'CITZ',
    label: 'Citizens Services',
  },
  {
    value: 'DBC',
    label: 'Drug Benefit Council',
  },
  {
    value: 'EAO',
    label: 'Environmental Assessment Office',
  },
  {
    value: 'EDUC',
    label: 'Education and Child Care',
  },
  {
    value: 'EMBC',
    label: 'Emergency Management',
  },
  {
    value: 'EMPR',
    label: 'Energy, Mines and Low Carbon Innovation',
  },
  {
    value: 'ENV',
    label: 'Environment and Climate Change Strategy',
  },
  {
    value: 'FIN',
    label: 'Finance',
  },
  {
    value: 'FLNR',
    label: 'Forests, Lands, Natural Resource',
  },
  {
    value: 'HLTH',
    label: 'Health',
  },
  {
    value: 'IRR',
    label: 'Indigenous Relations & Reconciliation',
  },
  {
    value: 'JEDC',
    label: 'Jobs, Economic Development and Innovation',
  },
  {
    value: 'LBR',
    label: 'Labour',
  },
  {
    value: 'LDB',
    label: 'Liquor Distribution Branch',
  },
  {
    value: 'MAH',
    label: 'Municipal Affairs and Housing',
  },
  {
    value: 'MCF',
    label: 'Children and Family Development',
  },
  {
    value: 'MMHA',
    label: 'Mental Health and Addictions',
  },
  {
    value: 'PSA',
    label: 'Public Service Agency',
  },
  {
    value: 'PSSG',
    label: 'Public Safety and Solicitor General',
  },
  {
    value: 'SDPR',
    label: 'Social Development and Poverty Reduction',
  },
  {
    value: 'TCA',
    label: 'Tangible Capital Assets',
  },
  {
    value: 'TRAN',
    label: 'Transportation and Infrastructure',
  },
  {
    value: 'HOUS',
    label: 'Ministry of Housing',
  },
];

export const clusterNames = [
  {
    id: 1,
    name: 'clab',
    humanFriendlyName: 'CLAB Calgary',
  },
  {
    id: 2,
    name: 'klab',
    humanFriendlyName: 'KLAB Kamloops',
  },
  {
    id: 3,
    name: 'silver',
    humanFriendlyName: 'Silver Kamloops',
  },
  {
    id: 4,
    name: 'gold',
    humanFriendlyName: 'Gold Kamloops',
  },
  {
    id: 5,
    name: 'golddr',
    humanFriendlyName: 'Gold (DR) Calgary',
  },
  {
    id: 6,
    name: 'klab2',
    humanFriendlyName: 'KLAB2 Kamloops',
  },
  {
    id: 7,
    name: 'emerald',
    humanFriendlyName: 'Emerald Hosting Tier',
  },
];

export const AGMinistries = ['AG', 'PSSG', 'EMBC', 'HOUS'];

export const productSorts = [
  {
    sortKey: 'updatedAt',
    humanFriendlyName: 'Product last update (new to old)',
    sortOrder: Prisma.SortOrder.desc,
  },
  {
    sortKey: 'updatedAt',
    humanFriendlyName: 'Product last update (old to new)',
    sortOrder: Prisma.SortOrder.asc,
  },
  {
    sortKey: 'name',
    humanFriendlyName: 'Product name (A-Z)',
    sortOrder: Prisma.SortOrder.asc,
  },
  {
    sortKey: 'name',
    humanFriendlyName: 'Product name (Z-A)',
    sortOrder: Prisma.SortOrder.desc,
  },
  {
    sortKey: 'description',
    humanFriendlyName: 'Product description (A-Z)',
    sortOrder: Prisma.SortOrder.asc,
  },
  {
    sortKey: 'description',
    humanFriendlyName: 'Product description (Z-A)',
    sortOrder: Prisma.SortOrder.desc,
  },
  {
    sortKey: 'licencePlate',
    humanFriendlyName: 'Product Licence Plate (A-Z)',
    sortOrder: Prisma.SortOrder.asc,
  },
  {
    sortKey: 'licencePlate',
    humanFriendlyName: 'Product Licence Plate (Z-A)',
    sortOrder: Prisma.SortOrder.desc,
  },
];

export const defaultCpuOptionsLookup: { [key: string]: string } = {
  CPU_REQUEST_0_5_LIMIT_1_5: '0.5 CPU Request, 1.5 CPU Limit',
  CPU_REQUEST_1_LIMIT_2: '1 CPU Request, 2 CPU Limit',
  CPU_REQUEST_2_LIMIT_4: '2 CPU Request, 4 CPU Limit',
  CPU_REQUEST_4_LIMIT_8: '4 CPU Request, 8 CPU Limit',
  CPU_REQUEST_8_LIMIT_16: '8 CPU Request, 16 CPU Limit',
  CPU_REQUEST_16_LIMIT_32: '16 CPU Request, 32 CPU Limit',
  CPU_REQUEST_32_LIMIT_64: '32 CPU Request, 64 CPU Limit',
  CPU_REQUEST_64_LIMIT_128: '64 CPU Request, 128 CPU Limit',
};

export const defaultMemoryOptionsLookup: { [key: string]: string } = {
  MEMORY_REQUEST_2_LIMIT_4: '2 GB Request, 4 GB Limit',
  MEMORY_REQUEST_4_LIMIT_8: '4 GB Request, 8 GB Limit',
  MEMORY_REQUEST_8_LIMIT_16: '8 GB Request, 16 GB Limit',
  MEMORY_REQUEST_16_LIMIT_32: '16 GB Request, 32 GB Limit',
  MEMORY_REQUEST_32_LIMIT_64: '32 GB Request, 64 GB Limit',
  MEMORY_REQUEST_64_LIMIT_128: '64 GB Request, 128 GB Limit',
  MEMORY_REQUEST_128_LIMIT_256: '128 Request, 256 GB Limit',
};

export const defaultStorageOptionsLookup: { [key: string]: string } = {
  STORAGE_1: '1 GB',
  STORAGE_2: '2 GB',
  STORAGE_4: '4 GB',
  STORAGE_16: '16 GB',
  STORAGE_32: '32 GB',
  STORAGE_64: '64 GB',
  STORAGE_128: '128 GB',
  STORAGE_256: '256 GB',
  STORAGE_512: '512 GB',
};

export const defaultProvisionedResourceValues = {
  cpu: 'CPU: 0.5 CPU Request, 1.5 CPU Limit',
  memory: 'Memory: 2 GB Request, 4 GB Limit',
  storage: 'Storage: 1 GB',
};

export const TEAM_SA_PREFIX = 'z_pltsvc-tsa-';

export const cpuOptions = Object.values(CpuQuotaEnum.enum);
export const memoryOptions = Object.values(MemoryQuotaEnum.enum);
export const storageOptions = Object.values(StorageQuotaEnum.enum);
