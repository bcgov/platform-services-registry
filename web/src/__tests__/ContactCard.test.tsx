//
// Copyright Â© 2020 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import { render } from '@testing-library/react';
import React from 'react';
import ContactCard from '../components/profileEdit/ContactCard';
import { ROLES } from '../constants';

test('matches the snapshot', () => {
  const stubPropPOId = '1';
  const stubPropPOFirstName = 'Jane';
  const stubPropPOLastName = 'Doe';
  const stubPropPOEmail = 'jane@example.com';
  const stubPropPORole = ROLES.PRODUCT_OWNER;
  const stubPropTLId = '2';
  const stubPropTLFirstName = 'Jim';
  const stubPropTLLastName = 'Doe';
  const stubPropTLEmail = 'jim@example.com';
  const stubPropTLRole = ROLES.TECHNICAL_LEAD;

  const { container } = render(
    <ContactCard
      contactDetails={[
        {
          id: stubPropPOId,
          firstName: stubPropPOFirstName,
          lastName: stubPropPOLastName,
          email: stubPropPOEmail,
          roleId: stubPropPORole,
        },
        {
          id: stubPropTLId,
          firstName: stubPropTLFirstName,
          lastName: stubPropTLLastName,
          email: stubPropTLEmail,
          roleId: stubPropTLRole,
        },
      ]}
    />,
  );

  expect(container).toMatchSnapshot();
});
