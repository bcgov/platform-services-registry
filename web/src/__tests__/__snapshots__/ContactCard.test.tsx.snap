// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <div
    class="css-z54kij"
  >
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Product Owner
      </h3>
      <p
        class="css-18ja79e"
      >
        This is typically the business owner of the application;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        Jane Doe
      </p>
      <p
        class="css-1guafq4"
      >
        jane@example.com
      </p>
      <p
        class="css-1guafq4"
      />
    </div>
    <div
      class="css-19crk1j"
    >
      <h3
        class="css-vurnku"
      >
        Technical Contact
      </h3>
      <p
        class="css-18ja79e"
      >
        This is typically the DevOps specialist;
      </p>
    </div>
    <div
      class="css-19crk1j"
    >
      <p
        class="css-1guafq4"
      >
        Jim Doe
      </p>
      <p
        class="css-1guafq4"
      >
        jim@example.com
      </p>
      <p
        class="css-1guafq4"
      />
    </div>
  </div>
</div>
`;
